src/merkle_tree/src/merkle_tree.cairo

```diff
@@ -168,36 +175,20 @@ pub impl MerkleTreeImpl<T, +HasherTrait<T>, +Copy<T>, +Drop<T>> of MerkleTreeTra
 fn compute_proof<T, +HasherTrait<T>, +Drop<T>>(
     mut nodes: Array<felt252>, mut hasher: T, index: u32, ref proof: Array<felt252>
 ) {
-    // Break if we have reached the top of the tree
-    if nodes.len() == 1 {
-        return;
+    if index % 2 == 0 {
+        proof.append(*nodes.at(index + 1));
+    } else {
+        proof.append(*nodes.at(index - 1));
     }
 
-    // If odd number of nodes, add a null virtual leaf
-    if nodes.len() % 2 != 0 {
-        nodes.append(0);
```

Comment:
Why is it better to have this in the wrapping function than here?
Does that mean that this method only accepts nodes with an even number of nodes?

Reply:
This avoid having it checked at every loop, it accepts node with odd len, it'll just add an empty node at the end then

Reply:
The wrapping `MerkleTreeTrait::compute_root` does.
But what about the helping function `compute_root`?
This one:
```
/// Helper function to compute a merkle proof of given leaves and at a given index.
/// # Arguments
/// * `nodes` - The sorted nodes.
/// * `index` - The index of the given.
/// * `hasher` - The hasher to use.
/// * `proof` - The proof array to fill.
fn compute_proof<T, +HasherTrait<T>, +Drop<T>>(
    mut nodes: Array<felt252>, mut hasher: T, index: u32, ref proof: Array<felt252>
) {
```

What if it is called with an odd number of nodes?

Reply:
it isn't public and can't (shouldn't) be called from any other fn. I tried removing that helper function but running into some issues. I could give it another try if you want

Reply:
Done